File: frontend/src/types/party.ts
import { z } from 'zod';

export const PartySchema = z.object({
  id: z.string().uuid(),
  transaction_id: z.string().uuid(),
  name: z.string(),
  role: z.string(),
  email: z.string(),
  phone: z.string().nullable().optional(),
  company: z.string().nullable().optional(),
  is_primary: z.boolean().default(false),
  notes: z.record(z.any()).nullable().optional(),
  created_at: z.string(),
  updated_at: z.string(),
});

export type Party = z.infer<typeof PartySchema>;

File: backend/app/schemas/party.py
from pydantic import BaseModel, ConfigDict
from typing import Any, Dict, Optional
from uuid import UUID
from datetime import datetime


class PartyBase(BaseModel):
    name: str
    role: str
    email: Optional[str] = None
    phone: Optional[str] = None
    company: Optional[str] = None
    is_primary: bool = False


class PartyCreate(PartyBase):
    pass


class PartyUpdate(BaseModel):
    """All fields optional for partial updates."""
    name: Optional[str] = None
    role: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    company: Optional[str] = None
    is_primary: Optional[bool] = None


class PartyResponse(PartyBase):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    transaction_id: UUID
    notes: Optional[Dict[str, Any]] = None
    created_at: datetime
    updated_at: datetime
